source/Ship.cpp

```diff
@@ -702,7 +713,10 @@ void Ship::FinishLoading(bool isNewInstance)
 		}
 		else
 			++it;
-		if(bay.side == Bay::INSIDE && bay.launchEffects.empty() && Crew())
+
+		if (removeBays)
+			it = bays.erase(it - 1);
```

Comment:
I don't think this is a good way to do this, not with the current syntax. Instead, go up to the point in `Ship::FinishLoading()` where it does `if(bays.empty() && !base->bays.empty())` (line 515 in this branch) and add `&& !removeBays` to the if statement so it doesn't copy the bays from the base if the variant specifies that it wants no bays.